SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp

```diff
@@ -0,0 +1,275 @@
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "PokemonSwSh/Programs/PokemonSwSh_EncounterHandler.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSwSh{
+    using namespace Pokemon;
+
+
+BerryFarmer2_Descriptor::BerryFarmer2_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonSwSh:BerryFarmer2",
+        STRING_POKEMON + " SwSh", "Date Spam - Berry Farmer 2",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonSwSh/DateSpam-BerryFarmer2.md",
+        "Farm berries using Feedback.",
+        FeedbackType::REQUIRED, false,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+
+
+
+BerryFarmer2::BerryFarmer2()
+    : REQUIRES_AUDIO(
+        html_color_text(
+            "<font size=4><b>Rustling detection uses sound. Make sure you have the correct audio input set.</b></font>",
+            COLOR_BLUE
+        )
+    )
+    , FETCH_ATTEMPTS(
+        "<b>Number of Fetch Attempts:</b>",
+        100000
+    )
+    , SAVE_ITERATIONS(
+        "<b>Save Every this Many Fetches:</b><br>(zero disables saving): ",
+        0
+    )
+    , m_advanced_options(
+        "<font size=4><b>Advanced Options:</b> You should not need to touch anything below here.</font>"
+    )
+    , EXIT_BATTLE_TIMEOUT(
+        "<b>Exit Battle Timeout:</b><br>After running, wait this long to return to overworld.",
+        TICKS_PER_SECOND,
+        "10 * TICKS_PER_SECOND"
+    )
+    , START_BATTLE_DELAY(
+        "<b>Start Battle Delay:</b><br>After a battle is detected, wait this long to flee.",
+        TICKS_PER_SECOND,
+        "15 * TICKS_PER_SECOND"
+    )
+    , RUSTLING_INTERVAL(
+        "<b>Rustling Interval:</b><br>How much time between two rustling sounds has to pass to be considered slow rustling in ms.",
+        1350
+    )
+    , RUSTLING_TIMEOUT(
+        "<b>Rustling Timeout:</b><br>Wait this many ticks to detect rustling.",
+        TICKS_PER_SECOND,
+        "400"
+    )
+    , SECONDARY_ATTEMPT_MASH_TIME(
+        "<b>Secondary attempt mash time:</b><br>Mash ZL this many ticks for secondary fetch attempts.",
+        TICKS_PER_SECOND,
+        "240"
+    )
+{
+    PA_ADD_OPTION(REQUIRES_AUDIO);
+    PA_ADD_OPTION(START_IN_GRIP_MENU);
+    PA_ADD_OPTION(FETCH_ATTEMPTS);
+    PA_ADD_OPTION(SAVE_ITERATIONS);
+
+    PA_ADD_STATIC(m_advanced_options);
+    PA_ADD_OPTION(EXIT_BATTLE_TIMEOUT);
+    PA_ADD_OPTION(START_BATTLE_DELAY);
+    PA_ADD_OPTION(RUSTLING_INTERVAL);
+    PA_ADD_OPTION(RUSTLING_TIMEOUT);
+    PA_ADD_OPTION(SECONDARY_ATTEMPT_MASH_TIME);
+}
+
+
+BerryFarmer2::Rustling BerryFarmer2::check_rustling(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
+    // wait some time in order to not detect rustling from previous fetch attempt
+    pbf_wait(context, 80);
+    context.wait_for_all_requests();
+
+    std::chrono::time_point initial_rustling_time = std::chrono::system_clock::now();
```

Comment:
could it be declared line 128 directly ?

Reply:
`Common/Cpp/Time.h` has `WallClock` and `current_time()` if you want to avoid writing the full chrono stuff.

Reply:
moved declaration to line 128 and changed to WallClock for readability.